### ISSUE TITLE
Table inline edit - dropdown

### ISSUE THREAD

**Issue by vconzola** [patternfly/patternfly-design#839]
There is a need in the OpenShift migration UI to be able to change the storage class of a given persistent volume (PV) within a table row. In the current implementation a dropdown (I guess technically, it's a PF select component.) with all of the available storage class options is displayed in each table row. See below.
![Screen Shot 2020-03-30 at 4 23 01 PM](https://user-images.githubusercontent.com/22823035/77958833-a9aacc00-72a3-11ea-9996-d96cc6746e77.png)
This is problematic because a "preferred" storage class is selected by default for each PV and we really don't want users messing with the selection unless they're sure they know what they're doing. A better design would be to display the "preferred" selection as text with the user having the option to edit it.

**Comment by mcarrano** [comment]
@vconzola I believe that our current inline edit approach would give us this capability. We just didn't create a demo of this. Can you confirm @mattnolting @seanforyou23 ?

cc @LHinson

**Comment by vconzola** [comment]
@mcarrano I thought the current inline edit only supports editing text fields. Hoping I'm wrong.  :-)

**Comment by mattnolting** [comment]
@mcarrano I believe @vconzola is correct. Only text fields are currently supported. @seanforyou23 can you confirm??

**Comment by katierik** [comment]
I have a use case where the user is looking at a long list of vulnerabilities to their infrastructure (5000+) and they can set a business risk and status for each. 
Ideally the user will be able to hover over the field in the table to show the edit affordance (instead of having it show 5000+ times!) and be able to edit the business risk value directly from the table. 

![Xnip2020-03-31_09-01-00](https://user-images.githubusercontent.com/23172038/78029179-297d7880-732e-11ea-977e-331a831b4ff0.jpg)

**Comment by mcarrano** [comment]
@mattnolting I'm a little surprised about that answer because I thought that the idea was that I could wrap any set of content in an inline edit container and toggle between read-only and editable states. For example I see that we could put radio buttons or check boxes into a table cell in this example: https://www.patternfly.org/v4/documentation/core/components/inlineedit#inline-edit-table-row So why not a dropdown select? But if this is not in fact supported out of the box, then I agree it's something we should enable.

Not sure about @katierik 's question. @seanforyou23 is it possible to wrap a single table cell like this? If not, it would be possible to place the entire row in edit mode and only make the Business risk field editable, I think.

**Comment by mattnolting** [comment]
@mcarrano you're correct, that's idea, but it hasn't been added to React yet.

**Comment by mattnolting** [comment]
@katierik you're saying you don't want to the see edit button, twice in each row, 5000 times. The idea being that there's an option to hide the edit button by default, on a single cell, then expose the button on hover. Is that correct?

**Comment by seanforyou23** [comment]
Catching up here - correct that inline edit pattern overall supports wrapping most any content types, but react-core doesn't yet have implementations for all these various scenarios. Currently, Table can do inline edit with basic text inputs. The plan was to add support for new controls as the need arises since it's a heavy lift in Table.

I'd caution against relying on only the hover state to show the edit buttons for usual reasons (hover state doesn't exist on mobile devices or with assistive technology). @katierik it sounds like you were talking about having the hover edit icon as _replacement_ for the persistent one instead of in addition to, is that right? So it wouldn't be 2x each table row, but still just 1x with the button in a slightly different location. Having the edit button available persistently for each row was by design (implementation-wise) because it provides an explicit path to navigate to the control that's unique. Here's a related issue that explains why this is important in a little more detail https://github.com/patternfly/patternfly-org/issues/1721 - as long as we can retain the discoverability we have currently, I'd be open to exploring alternate implementation solutions.

I think it would be good to open a dedicated issue for adding dropdown select support for inline edit in Table, and maybe address the edit button placement piece separately?

**Comment by mcarrano** [comment]
I think that all makes sense @seanforyou23 . Should I just transfer this issue over to React then? I don't think there's further design or core work that needs to be done unless I'm missing something.

**Comment by seanforyou23** [comment]
Yes for the select support just transferring this over to React would be great! We can follow-up and discuss showing the edit control on hover for @katierik's In another thread if that's ok.
